src/main/java/com/baidu/hugegraph/util/CollectionUtil.java

```diff
@@ -108,10 +114,12 @@ public static boolean allUnique(Collection<?> collection) {
      * @return sub-set of original set [from, to)
      */
     public static <T> Set<T> subSet(Set<T> original, int from, int to) {
-        List<T> list = new ArrayList<>(original);
-        if (to == -1) {
+        E.checkArgument(from >= 0,
+                        "Invalid from parameter of subSet(): %s", from);
+        if (to < 0) {
```

Comment:
"if (to < 0)" => "if (to < from)". IllegalArgumentException will be thrown if  to < from。